components/proxy/src/test/kotlin/com/hotels/styx/ServerConfigSchemaTest.kt

```diff
@@ -163,6 +163,14 @@ class ServerConfigSchemaTest : DescribeSpec({
         }
     }
 
+    describe("Incorrect configuration") {
+        it("Detects unknown configuration options") {
+            validateServerConfiguration(yamlConfig(minimalConfig + """
+                abc: 1
+            """.trimIndent())) shouldBe (Optional.of("Unexpected field: '.abc'"))
```

Comment:
Why is the field name prefixed with `.` - shouldn't that be just for delimiting between parts of the name?

Reply:
Okay. This allows us to avoid a special case in the code.

But let me have another look into this. I also would like to improve the messages.

Reply:
I have addressed this issue.